spec/01-unit/05-utils_spec.lua

```diff
@@ -229,11 +229,12 @@ describe("Utils", function()
         local str = utils.encode_args {
           array = {"hello, world"},
           hash = { answer = 42 },
+          nested = { array = { "one", "two" } },
           hello = "world",
           falsy = false,
           ["multiple values"] = true
         }
-        assert.equal("array%5b1%5d=hello%2c%20world&falsy=false&hash%2eanswer=42&hello=world&multiple%20values=true", str)
+        assert.equal("array%5b1%5d=hello%2c%20world&falsy=false&hash%2eanswer=42&hello=world&multiple%20values=true&nested%2earray%5b1%5d=one&nested%2earray%5b2%5d=two", str)
```

Comment:
We should also add tests asserting that the `raw` argument is still respected for nested table values.

Reply:
Another _unit_ test case that seems to be missing is nested hash table cases (non-integer-indexed tables) and even maybe mixed tables.

Reply:
Other missing test case: nested tables with `no_array_indexes`

Reply:
I have added a bunch of tests - `array`, `hash`, `array_array`, `hash_hash`, `array_hash`, `hash_array` and `hybrid`.